apps/backoffice/src/app/api/lib/authorized-cidrs.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  HTTP_STATUS_INTERNAL_SERVER_ERROR,
+  HTTP_STATUS_NOT_FOUND
+} from "@/config/constants";
+import { SubscriptionCIDRsModel } from "@pagopa/io-functions-commons/dist/src/models/subscription_cidrs";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { NextResponse } from "next/server";
+import { BackOfficeUser } from "../../../../types/next-auth";
+
+export const retrieveManageKeyCIDRs = (
+  subscriptionCIDRsModel: SubscriptionCIDRsModel
+) => (backOfficeUser: BackOfficeUser) =>
+  pipe(
+    subscriptionCIDRsModel.findLastVersionByModelId([
+      backOfficeUser.parameters.subscriptionId as NonEmptyString
+    ]),
+    TE.mapLeft(error =>
+      NextResponse.json(
+        {
+          title: "ManageKeyCIDRsError",
+          status: HTTP_STATUS_INTERNAL_SERVER_ERROR,
+          detail: JSON.stringify(error)
+        },
+        { status: HTTP_STATUS_INTERNAL_SERVER_ERROR }
+      )
+    ),
+    TE.chainW(
+      TE.fromOption(() =>
+        NextResponse.json(
+          {
+            title: "ManageKeyCIDRsNotFound",
+            status: HTTP_STATUS_NOT_FOUND,
+            detail: "Manage Key CIDRs not found"
+          },
+          { status: HTTP_STATUS_NOT_FOUND }
+        )
+      )
+    ),
+    TE.map(response =>
+      NextResponse.json({
+        cidrs: response.cidrs
+      })
+    ),
+    TE.toUnion
+  );
```

Comment:
**suggestion (non-blocking):** move the NextResponse to the caller (request handler).

I suggest to remove from this lib/util the responsibility for returning the HTTP response, but instead return a structured error object to the caller (the NextJS request handler), so that it is able to create the correct HTTP response

Reply:
To make the controllers easily testable, I expressly chose in the design phase of these APIs not to write any logic within the methods of the various route.ts of next.js.
This choice was made because of the framework's lack of dependency injection, which makes it difficult to write unit tests.

The controller:
- obtains the dependency instances necessary for execution
- calls the corresponding a method that contains the controller logic by injecting the dependencies and request data.

The method in question executes:
- a call to a business method (whose dependencies are injected), which fetches the data from cosmos
- based on the result from cosmos composes the response, which is immediately returned in the controller